contracts/access/manager/AccessManager.sol

```diff
@@ -0,0 +1,564 @@
+     *
+     * Requirements:
+     *
+     * - the caller must be in the group's admins
+     *
+     * Emit a {GroupRevoked} event
+     */
+    function revokeGroup(uint256 groupId, address account) public virtual onlyGroup(getGroupAdmin(groupId)) {
+        _revokeGroup(groupId, account);
+    }
+
+    /**
+     * @dev Renounce group permissions for the calling account, with immediate effect.
+     *
+     * Requirements:
+     *
+     * - the caller must be `callerConfirmation`.
+     *
+     * Emit a {GroupRevoked} event
+     */
+    function renounceGroup(uint256 groupId, address callerConfirmation) public virtual {
+        if (callerConfirmation != _msgSender()) {
+            revert AccessManagerBadConfirmation();
+        }
+        _revokeGroup(groupId, callerConfirmation);
+    }
+
+    /**
+     * @dev Set the execution delay for a given account in a given group. This update is not immediate and follows the
+     * delay rules. For example, If a user currently has a delay of 3 hours, and this is called to reduce that delay to
+     * 1 hour, the new delay will take some time to take effect, enforcing that any operation executed in the 3 hours
+     * that follows this update was indeed scheduled before this update.
+     *
+     * Requirements:
+     *
+     * - the caller must be in the group's admins
+     *
+     * Emit a {GroupExecutionDelayUpdate} event
+     */
+    function setExecuteDelay(
+        uint256 groupId,
+        address account,
+        uint32 newDelay
+    ) public virtual onlyGroup(getGroupAdmin(groupId)) {
+        _setExecuteDelay(groupId, account, newDelay, false);
+    }
+
+    /**
+     * @dev Change admin group for a given group.
+     *
+     * Requirements:
+     *
+     * - the caller must be a global admin
+     *
+     * Emit a {GroupAdminChanged} event
+     */
+    function setGroupAdmin(uint256 groupId, uint256 admin) public virtual onlyGroup(ADMIN_GROUP) {
+        _setGroupAdmin(groupId, admin);
+    }
+
+    /**
+     * @dev Change guardian group for a given group.
+     *
+     * Requirements:
+     *
+     * - the caller must be a global admin
+     *
+     * Emit a {GroupGuardianChanged} event
+     */
+    function setGroupGuardian(uint256 groupId, uint256 guardian) public virtual onlyGroup(ADMIN_GROUP) {
+        _setGroupGuardian(groupId, guardian);
+    }
+
+    /**
+     * @dev Update the .
+     *
+     * Requirements:
+     *
+     * - the caller must be a global admin
+     *
+     * Emit a {GroupGrantDelayChanged} event
+     */
+    function setGrantDelay(uint256 groupId, uint32 newDelay) public virtual onlyGroup(ADMIN_GROUP) {
+        _setGrantDelay(groupId, newDelay, false);
+    }
+
+    /**
+     * @dev Internal version of {grantGroup} without access control.
+     *
+     * Emit a {GroupGranted} event
+     */
+    function _grantGroup(uint256 groupId, address account, uint32 grantDelay, uint32 executionDelay) internal virtual {
```

Comment:
I've been thinking of a few scenarios where I think it makes sense to disallow `address(this)`. For example, if someone decides to make the manager itself its initial address.

Another thing to consider is that any user with permissions to call the `relay` function of the manager inherits its permission. We can just warn users since it's low likely imo but I think it's easier to:

```suggestion
        function _grantGroup(uint256 groupId, address account, uint32 grantDelay, uint32 executionDelay) internal virtual {
        if (account == address(this)) {
            revert AccessManagerInvalidAdmin(address(this));
        }
```

Reply:
Calling the relay function is not obvious though:
- the target and selector must have their permission set
- the caller (`_msgSender`) must have the corresponding permission
- if there is a setback, the operation must have been scheduled.

That means that you can only relay to targets that you are explicitelly allowed to relay. So the admin would have to give the user to call relay through the relay, by setting some specific permission to the relay function ... and then the manager should have some permission (which you want to forbid).

An equally reasonnable option would be to prevent `target == address(this)` in `_setFunctionAllowedGroup`, (and also to prevent changing the mode of the manager in `_setContractMode`)

Reply:
we have two conversations in parallele. see: https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4416#discussion_r1280113464

Reply:
I agree we should consider preventing `target == address(this)` in `_setFunctionAllowedGroup` but I'm not convinced yet. I agree the likelihood of this is very granting access to the `relay` function and also having the manager with permissions is low likelihood but I think we're recommending the `relay` function as an adapter replacement, aren't we?

In such case the recommendation we're making is to give the manager permissions so that it can relay access managed functions.

Reply:
Addressed in https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4520/commits/a6ba3a6a62d3f6892bc9659157fcd84ec5f9b147